packages/shell-api/src/collection.ts

```diff
@@ -999,7 +999,7 @@ export default class Collection extends ShellApiClass {
     this._emitCollectionApiCall('createIndex', { keys, options });
 
     const spec = { ...this._database._baseOptions, ...options, key: keys };
-    return await this._mongo._serviceProvider.createIndexes(this._database._name, this._name, [spec]);
+    return await this._mongo._serviceProvider.createIndexes(this._database._name, this._name, [spec], this._database._baseOptions);
```

Comment:
@aherlihy Should this be 
```suggestion
    return await this._mongo._serviceProvider.createIndexes(this._database._name, this._name, [spec], options, this._database._baseOptions);
```

? Or should the options really be passed as part of the index spec?

Reply:
Looks like node.js driver expects only `session` in `options` and other properties in spec.
So maybe more refactoring and unification is required in core

Reply:
We should definitely add more docs/tests since this is confusing,  but there is a bug in the original code. That being said Ludmilla's more correct for this one, although we need to tweak the entire method a bit (which can happen in a different ticket).

The shell API allows users to pass index-specific options via the `options` argument, which are described here: https://docs.mongodb.com/manual/reference/method/db.collection.createIndex/index.html#options-for-all-index-types

The node driver it seems, can take those indexes-specific options as part of the index spec or as a second argument. Right now it looks like we are adding it to the spec (which is why the last two args of `ServiceProvider.createIndexes` were skipped). 

But the session option cannot be set via the index spec and instead needs to be sent via `options`, so we had a bug where we were sending the `session` option via the index spec which would get ignored.  

Note there are no `DatabaseOptions` to be sent at all so the last argument to `service provider.createIndexes` should still be empty or {}. (`this._database._baseOptions` are command options for all commands associated with that database, and not actual `DatabaseOptions`).

I think a better alternative for everyone is to try separating out the options from the index spec. So the method would look like: 
```js
@returnsPromise
  async createIndex(
    keys: Document,
    options: Document = {}
  ): Promise<Document> {
    assertArgsDefined(keys);
    if (typeof options !== 'object' || Array.isArray(options)) {
      throw new MongoshInvalidInputError('The "options" argument must be an object.');
    }
    this._emitCollectionApiCall('createIndex', { keys, options });

    return await this._mongo._serviceProvider.createIndexes(this._database._name, this._name, [{ key: keys }], { ...this._database._baseOptions, ...options});
  }
```
but this change should come with tests and such so I can make it separately: https://jira.mongodb.org/browse/MONGOSH-478

Sorry for the super long response!